src/Microsoft.Identity.Client/Cache/CacheFallbackOperations.cs

```diff
@@ -237,7 +234,7 @@ internal static class CacheFallbackOperations
         public static List<MsalRefreshTokenCacheItem> GetAllAdalEntriesForMsal(
             ICoreLogger logger,
             ILegacyCachePersistence legacyCachePersistence,
-            ISet<string> environmentAliases,
+            IEnumerable<string> environmentAliases,
```

Comment:
IEnumerable is more appropriate for aliases because sometimes we are ordering aliases, e.g. use preferred cache first

Reply:
So, an ISet is an IEnumerable.  The one thing I like about Set semantics with scopes is it guarantees there are no duplicates, which IEnumerable by itself does not.  thoughts?

Reply:
Yes, but it seems to be a bit too much work to use ISet everywhere - DataContract does not support this, semantics are different because you have to specify CultureInfo in the ctor etc. I could transform the IEnumerable to an ISet, but since the values we get from AAD won't have duplicates, IEnumerable seems fine, especially being readonly. I'll have another go at this in the second round of refactoring, where I introduce hardcoded values.